packages/dds/matrix/src/matrix.ts

```diff
@@ -634,20 +632,14 @@ export class SharedMatrix<T = any>
 	}
 
 	private rebasePosition(
-		// eslint-disable-next-line import/no-deprecated
-		client: Client,
-		pos: number,
-		referenceSequenceNumber: number,
+		client: PermutationVector,
 		localSeq: number,
+		handle: number,
 	): number | undefined {
-		const { clientId } = client.getCollabWindow();
-		const { segment, offset } = client.getContainingSegment(
-			pos,
-			{ referenceSequenceNumber, clientId: client.getLongClientId(clientId) },
-			localSeq,
-		);
-		if (segment === undefined || offset === undefined) {
-			return;
+		const { segment, offset } = client.handleToSegment(handle);
```

Comment:
Old code called 'MergeTree.getContainingSegment(..)' to resolve a past row/col _position_ to a segment + offset

New code calls 'handleToSegment()', which linearly scans the MergeTree to locate the row/col _handle_ to find the segment + offset.